### PR TITLE
06/get/api/articles/article id/comments

### PR DESCRIPTION
**Pull request by raulmartinasc**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -79,14 +79,70 @@ describe("NC-News", () => {
...
       return request(app)
         .get("/notARoute")
         .expect(404)
         .then(({ body }) => {
           expect(body.msg).toBe("Path not found");
         });
     });
+    test("status:404-/api/articles/99999 responds with an error message when that Id does not exist", () => {
+      return request(app)
+        .get("/api/articles/99999")
```
Comment (itskatherine) [inline __tests__/app.test.js:146]: This endpoint should have "/comments" on the end shouldn't it? Or is this for ticket 5? - (get an article by ID)

File: __tests__/app.test.js [hunk 2]
```diff
@@ -79,14 +79,70 @@ describe("NC-News", () => {
...
+          expect(body.article).toHaveProperty("topic", expect.any(String));
+          expect(body.article).toHaveProperty("created_at", expect.any(String));
+          expect(body.article).toHaveProperty("votes", expect.any(Number));
+          expect(body.article).toHaveProperty(
+            "article_img_url",
+            expect.any(String)
+          );
+        });
+    });
+  });
```
Comment (itskatherine) [inline __tests__/app.test.js:110]: These two above tests are for a previous ticket (ticket 5) so shouldn't be included on this ticket

File: __tests__/app.test.js [hunk 3]
```diff
@@ -79,14 +79,70 @@ describe("NC-News", () => {
...
+        .get("/api/articles/99999")
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Article not found");
+        });
+    });
+    test("status 400 /api/articles/notAnID, responds with an error message when is an invalid Id", () => {
+      return request(app)
+        .get("/api/articles/notAnID")
+        .expect(400)
```
Comment (itskatherine) [inline __tests__/app.test.js:155]: Same for this, confused which ticket we're working on here

File: __tests__/app.test.js [hunk 4]
```diff
@@ -79,14 +79,70 @@ describe("NC-News", () => {
...
+  describe("GET /api/articles/:article_id/comments", () => {
+    test("It should respond with an array of comments", () => {
+      return request(app)
+        .get("/api/articles/1/comments")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.comments).toBeInstanceOf(Array);
+          expect(body.comments).toHaveLength(11);
+        });
+    });
```
Comment (itskatherine) [inline __tests__/app.test.js:132]: This is a good test for this ticket, though we'll need a few more for the endpoint /api/articles/:article_id/comments
400 - invalid article ID
404 - article not found
We'll also need to test that the comments are in date order (using jest-sorted)
We'll also need to test that when we look for the comments where the article exists but there are no comments, we get an empty array rather than a 404 error 🔍

File: db/models.js [hunk 5]
```diff
@@ -23,3 +23,24 @@ exports.fetchAllArticles = () => {
...
+      } else {
+        return article;
+      }
+    });
+};
+
+exports.selectCommentsByArticleId = (article_id) => {
+  return db
+    .query("SELECT * FROM comments WHERE article_id = $1;", [article_id])
+    .then(({ rows }) => {
```
Comment (itskatherine) [inline db/models.js:43]: Will need to add more functionality to this function to pass the tests I listed above 🧪

**Review by itskatherine — CHANGES_REQUESTED** [review]
Hi Raul, need a bit more functionality for the /articles/:article_id/comments endpoint which I've outlined here 🙂 
Could you add these changes and resubmit the PR once you have pls?

File: __tests__/app.test.js [hunk 6]
```diff
@@ -109,16 +109,29 @@ describe("NC-News", () => {
...
   describe("GET /api/articles/:article_id/comments", () => {
-    test("It should respond with an array of comments", () => {
+    test("It should respond with an array of comments ordered by date", () => {
       return request(app)
         .get("/api/articles/1/comments")
         .expect(200)
         .then(({ body }) => {
           expect(body.comments).toBeInstanceOf(Array);
           expect(body.comments).toHaveLength(11);
+          const ArrayOfDates = body.comments.map((comment) => {
```
Comment (itskatherine) [inline __tests__/app.test.js:119]: Avoid capitalised variable names, they tend to denote classes 🕵️

**Review by itskatherine — APPROVED** [review]
Great Raul, have a look at my very minor comment, then good to merge! 💪
